lib/ActivityStream.jsm

```diff
@@ -202,33 +202,34 @@ const PREFS_CONFIG = new Map([
     title: "Does the user allow CFR recommendations?",
     value: true,
   }],
-  ["asrouter.messageProviders", {
-    title: "Configuration for ASRouter message providers",
-
-    /**
-     * Each provider must have a unique id and a type of "local" or "remote".
-     * Local providers must specify the name of an ASRouter message provider.
-     * Remote providers must specify a `url` and an `updateCycleInMs`.
-     * Each provider must also have an `enabled` boolean.
-     */
-    value: JSON.stringify([{
+  ["asrouter.providers.onboarding", {
+    title: "Does the user allow CFR recommendations?",
+    value: JSON.stringify({
       id: "onboarding",
       type: "local",
       localProvider: "OnboardingMessageProvider",
       enabled: true,
-    }, {
+    }),
+  }],
+  ["asrouter.providers.snippets", {
+    title: "Does the user allow CFR recommendations?",
+    value: JSON.stringify({
       id: "snippets",
       type: "remote",
       url: "https://snippets.cdn.mozilla.net/%STARTPAGE_VERSION%/%NAME%/%VERSION%/%APPBUILDID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/%OS_VERSION%/%DISTRIBUTION%/%DISTRIBUTION_VERSION%/",
       updateCycleInMs: ONE_HOUR_IN_MS * 4,
-      enabled: false,
-    }, {
+      enabled: UpdateUtils.getUpdateChannel(true) === "nightly",
+    }),
+  }],
+  ["asrouter.providers.cfr", {
+    title: "Does the user allow CFR recommendations?",
+    value: JSON.stringify({
       id: "cfr",
       type: "local",
       localProvider: "CFRMessageProvider",
-      enabled: IS_NIGHTLY_OR_UNBRANDED_BUILD,
-      cohort: IS_NIGHTLY_OR_UNBRANDED_BUILD ? "nightly" : "",
-    }]),
+      frequency: {custom: [{period: "daily", cap: 1}]},
+      enabled: true,
```

Comment:
Why is the frequency here now?

Reply:
Before it was in the experiment definition here: https://gist.github.com/k88hudson/8e1fb4a7a84cd521832aaf36cb6f62e3 now that it's getting shipped to production, it needs to be in the message definition.